controllers/pod_models.go

```diff
@@ -1250,6 +1256,39 @@ func NormalizeClusterSpec(spec *fdbtypes.FoundationDBClusterSpec, options Deprec
 	return nil
 }
 
+// GetPodDisruptionBudgets generates the disruption budgets that we need to
+// configure for a cluster.
+func GetPodDisruptionBudgets(cluster *fdbtypes.FoundationDBCluster) ([]policyv1beta1.PodDisruptionBudget, error) {
+	if !*cluster.Spec.EnablePodDisruptionBudget {
+		return nil, nil
+	}
+	budgets := make([]policyv1beta1.PodDisruptionBudget, 0, len(fdbtypes.ProcessClasses))
+	processCounts, err := cluster.GetProcessCountsWithDefaults()
+	if err != nil {
+		return nil, err
+	}
+	processCountsMap := processCounts.Map()
+	for _, processClass := range fdbtypes.ProcessClasses {
+		count := processCountsMap[processClass]
+		if count > 0 {
+			metadata := getObjectMetadata(cluster, nil, processClass, "")
+			metadata.OwnerReferences = buildOwnerReference(cluster.TypeMeta, cluster.ObjectMeta)
+			metadata.Name = fmt.Sprintf("%s-%s", cluster.Name, processClassSanitizationPattern.ReplaceAllString(string(processClass), "-"))
+			minAvailable := intstr.FromInt(count - 1)
```

Comment:
I think for now this is okay but in the end we could be smarter and use `DesiredFaultTolerance` for SS and maybe a better default for `stateless` processes.